app/.storybook/main.ts

```diff
@@ -3,16 +3,25 @@
  * Handles things like config for location of story files and managing presets (which configure webpack and babel).
  * @see https://storybook.js.org/docs/configurations/default-config/
  */
+import type { StorybookConfig } from "@storybook/core-common";
 
-const nextConfig = require("../next.config");
+import nextConfig from "../next.config";
 
-module.exports = {
+const config: StorybookConfig = {
   stories: ["../stories/**/*.stories.@(mdx|js|jsx|ts|tsx)"],
   addons: ["@storybook/addon-essentials", "storybook-react-i18next"],
   framework: "@storybook/react",
   core: {
-    // Use webpack5 instead of webpack4.
-    builder: "webpack5",
+    builder: {
+      name: "webpack5",
+      options: {
+        // Cache build output between runs, to speed up subsequent startup times
+        fsCache: true,
+        // Applies in development mode. Storybook will start up faster, at the cost
+        // of slightly slower browsing time when you navigate to another story.
+        lazyCompilation: true,
```

Comment:
Does lazy compilation mean basically that a story will only compile when the browser requests it to load? 

I was reading this: https://storybook.js.org/blog/storybook-lazy-compilation-for-webpack/ and the explanation felt slightly different than this https://webpack.js.org/configuration/experiments/#experimentslazycompilation so just wanted to understand.

Reply:
> Does lazy compilation mean basically that a story will only compile when the browser requests it to load?

Yea this was my understanding based on their post. I agree the webpack doc is kinda confusing.

> I feel like for the claim status prototype for example we might not want it if we anticipate folks would try every single claim status option

I don't think this is an issue since lazy compilation only impacts how stories compile during local development (`npm run storybook`), but for the published Storybook, those are all compiled at build time (`npm run storybook-build`)

Reply:
oh right! thank you sounds good!